core/src/main/java/org/jdbi/v3/core/mapper/reflect/internal/BeanPropertiesFactory.java

```diff
@@ -135,10 +98,33 @@ public T build() {
         static class BeanPojoProperty<T> implements PojoProperty<T> {
             final PropertyDescriptor descriptor;
             final QualifiedType<?> qualifiedType;
+            final ConcurrentMap<Class<?>, Optional<Annotation>> annoCache = new ConcurrentHashMap<>();
+            final Optional<Function<Object, Object>> getter;
```

Comment:
I'd recommend against using Optional<> as a field type, it is intended for return method type. as described here http://blog.joda.org/2014/11/optional-in-java-se-8.html